s2n.mk

```diff
@@ -181,84 +181,19 @@ bindir ?= $(exec_prefix)/bin
 libdir ?= $(exec_prefix)/lib64
 includedir ?= $(exec_prefix)/include
 
-try_compile = $(shell $(CC) $(CFLAGS) -c -o tmp.o $(1) > /dev/null 2>&1; echo $$?; rm tmp.o > /dev/null 2>&1)
+feature_probe = $(shell $(CC) $(CFLAGS) $(shell cat $(S2N_ROOT)/tests/features/$(1).flags) -c -o tmp.o $(S2N_ROOT)/tests/features/$(1).c > /dev/null 2>&1 && echo "-D$(1)"; rm tmp.o > /dev/null 2>&1)
 
-# Determine if execinfo.h is available
-TRY_COMPILE_EXECINFO := $(call try_compile,$(S2N_ROOT)/tests/features/execinfo.c)
-ifeq ($(TRY_COMPILE_EXECINFO), 0)
-	DEFAULT_CFLAGS += -DS2N_STACKTRACE
-endif
-
-# Determine if cpuid.h is available
-TRY_COMPILE_CPUID := $(call try_compile,$(S2N_ROOT)/tests/features/cpuid.c)
-ifeq ($(TRY_COMPILE_CPUID), 0)
-	DEFAULT_CFLAGS += -DS2N_CPUID_AVAILABLE
-endif
+FEATURES := $(notdir $(patsubst %.c,%,$(wildcard $(S2N_ROOT)/tests/features/*.c)))
+SUPPORTED_FEATURES := $(foreach feature,$(FEATURES),$(call feature_probe,$(feature)))
+CFLAGS += $(SUPPORTED_FEATURES)
+DEFAULT_CFLAGS += $(SUPPORTED_FEATURES)
+CPPCFLAGS += $(SUPPORTED_FEATURES)
```

Comment:
How were CFLAGS AND CPPCFLAGS being set before? Were they being set before?

Reply:
They weren't! Meaning anything using those wasn't getting any feature flags. I ran into this with the fuzz test while implementing #3798.